quickwit/quickwit-indexing/src/source/queue_sources/message.rs

```diff
@@ -344,4 +347,51 @@ mod tests {
             panic!("Expected skippable error");
         }
     }
+
+    #[test]
+    fn test_uri_from_s3_notification_url_decode() {
+        let test_message = r#"
+        {
+            "Records": [
+                {
+                "eventVersion": "2.1",
+                "eventSource": "aws:s3",
+                "awsRegion": "us-west-2",
+                "eventTime": "2021-05-22T09:22:41.789Z",
+                "eventName": "ObjectCreated:Put",
+                "userIdentity": {
+                    "principalId": "AWS:AIDAJDPLRKLG7UEXAMPLE"
+                },
+                "requestParameters": {
+                    "sourceIPAddress": "127.0.0.1"
+                },
+                "responseElements": {
+                    "x-amz-request-id": "C3D13FE58DE4C810",
+                    "x-amz-id-2": "FMyUVURIx7Zv2cPi/IZb9Fk1/U4QfTaVK5fahHPj/"
+                },
+                "s3": {
+                    "s3SchemaVersion": "1.0",
+                    "configurationId": "testConfigRule",
+                    "bucket": {
+                        "name": "mybucket",
+                        "ownerIdentity": {
+                            "principalId": "A3NL1KOZZKExample"
+                        },
+                        "arn": "arn:aws:s3:::mybucket"
+                    },
+                    "object": {
+                        "key": "hello::world::logs.json",
```

Comment:
isn't this where it is supposed to be URL encoded?

Reply:
from what I remember, this test fails without url decoding

Reply:
Have you checked that this test fails without the URL decoding? I don't think it does,`hello::world::logs.json` is not url encoded.